### PR TITLE
Emend Photo Card max-photo-size-warning.

### PR DESCRIPTION
**Pull request by khodgkinson-he**
### Types of changes

-   [X] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

Upload Photographs widget changed to display max-photo-size-warning containing dynamic "maximum size of any single file" value corresponding to the Max File Size as set on the widget manager / db.

image

### Issues Solved

#10653

### CODE REVIEW

File: arches/app/templates/views/components/photo-workbench.htm [hunk 1]
```diff
@@ -114,7 +114,8 @@ <h4>
                                 <span data-bind="text: $root.translations.addingPhotosOptional"></span>
                             </p>
                             <p>
-                                <span data-bind="text: $root.translations.maxPhotoSizeWarning"></span>
+                                <span data-bind="text: $root.translations.maxPhotoSizeWarning + ' : '"></span>
+                                <span data-bind="text: maxFilesize() + ' MB'"></span>
```
Comment (jacobtylerwalls) [inline arches/app/templates/views/components/photo-workbench.htm:118]: Nice catch!

This won't give the translator much flexibility about where to put the size in the sentence, and probably won't work for RTL. I'm wondering if you considered including a placeholder in the string (like `{maxFilesize}`), and then replacing the placeholder in the javascript?

Comment (khodgkinson-he) [reply arches/app/templates/views/components/photo-workbench.htm]: Absolutely.. ..makes good sense to me!  

Amended to use placeholder!  ^_^

File: arches/app/templates/javascript.htm [hunk 2]
```diff
@@ -524,7 +524,7 @@
         uploaded-files='{% trans "Uploaded Files" as uploadedFiles %} "{{ uploadedFiles|escapejs }}"'
         allowed-document-formats='{% trans "Allowed document formats:" as allowedDocumentFormats %} "{{ allowedDocumentFormats|escapejs }}"'
         max-file-size-warning='{% trans "You may upload as many documents as you wish, but the maximum size of any single file is" as maxFileSizeWarning %} "{{ maxFileSizeWarning|escapejs }}"'
-        max-photo-size-warning='{% trans "You may upload as many photos as you wish, but the maximum size of any single file is 8MB." as maxPhotoSizeWarning %} "{{ maxPhotoSizeWarning|escapejs }}"'
+        max-photo-size-warning='(maxFilesize) => {return {% trans "You may upload as many photos as you wish, but the maximum size of any single file is ${maxFilesize} MB" as maxPhotoSizeWarning %} `{{ maxPhotoSizeWarning|escapejs }}` }'
```
Comment (jacobtylerwalls) [inline arches/app/templates/javascript.htm:527]: Figure we should keep the punctuation
```suggestion
        max-photo-size-warning='(maxFilesize) => {return {% trans "You may upload as many photos as you wish, but the maximum size of any single file is ${maxFilesize} MB." as maxPhotoSizeWarning %} `{{ maxPhotoSizeWarning|escapejs }}` }'
```

File: arches/app/media/js/views/components/cards/photo-gallery-card.js [hunk 3]
```diff
@@ -49,6 +49,16 @@ define([
 
         this.fileListNodeId = getfileListNode();
 
+        this.maxFilesize = ko.computed(function(){
+            var mfs = "Missing maxFilesize"
```
Comment (jacobtylerwalls) [inline arches/app/media/js/views/components/cards/photo-gallery-card.js:53]:
```suggestion
            var mfs = "Missing maxFilesize";
```

File: arches/app/media/js/views/components/cards/photo-gallery-card.js [hunk 4]
```diff
@@ -49,6 +49,16 @@ define([
...
+            self.card.widgets().forEach(function(widget){
+                if (widget.node_id() === self.fileListNodeId) {
+                    mfs = widget.config.maxFilesize();
+                }
+            });
+            return mfs;
```
Comment (jacobtylerwalls) [inline arches/app/media/js/views/components/cards/photo-gallery-card.js:59]: What should we show where there is no max file size? With this PR we show:

Screenshot 2024-03-08 at 4 14 05 PM

**Review by jacobtylerwalls — COMMENTED** [review]
Great! One question about when the max is missing.

File: arches/app/media/js/views/components/cards/photo-gallery-card.js [hunk 5]
```diff
@@ -49,6 +49,16 @@ define([
 
         this.fileListNodeId = getfileListNode();
 
+        this.maxFilesize = ko.computed(function(){
+            var mfs = "Missing maxFilesize";
+            self.card.widgets().forEach(function(widget){
+                if (widget.node_id() === self.fileListNodeId) {
+                    mfs = widget.config.maxFilesize();
```
Comment (khodgkinson-he) [inline arches/app/media/js/views/components/cards/photo-gallery-card.js:56]:
```suggestion
                    mfs = widget.config.maxFilesize() || "undefined";
```

Quite so! If undefined, let's inform the user. ^_^

image

Comment (jacobtylerwalls) [reply arches/app/media/js/views/components/cards/photo-gallery-card.js:56]: Something symbolic like `--` might be even better.

Comment (chiatt) [reply arches/app/media/js/views/components/cards/photo-gallery-card.js:56]: Maybe that entire sentence doesn't need to be displayed if there is no max size?

File: arches/app/media/js/views/components/cards/photo-gallery-card.js (hunk 4 above)
Comment (khodgkinson-he) [reply arches/app/media/js/views/components/cards/photo-gallery-card.js:59]: ..thanks @jacobtylerwalls!  

If undefined, let's inform the user.. Suggestion below..

File: arches/app/media/js/views/components/cards/photo-gallery-card.js (hunk 5 above)
Comment (chiatt) [reply arches/app/media/js/views/components/cards/photo-gallery-card.js:56]: I think we should go with @jacobtylerwalls `--` suggestion. That way we can keep the message and conveys to the user that no max is enforced.

**Comment by khodgkinson-he** [comment]
image

That looks to convey the meaning well, and should be most amenable to translations.

Thank you for all your help!  ^_^

**Review by jacobtylerwalls — APPROVED** [review]
Thanks!

**Pull request merged**